tests/test_integration.py

```diff
@@ -37,124 +88,184 @@ def test_vendor_dry_run(python_main_defaults_environment):
-    test_path = tmp_path_factory.mktemp("full-runthrough")
-    os.chdir(test_path)
 
+# full conda vendor call. will use mamba to solve if that's available, otherwise
+# uses conda
+@pytest.mark.integration
+def test_vendor_complete(make_env):
+    solver = "mamba" if _has_mamba() else "conda"
+    env = make_env(environments["minimal"])
     runner = CliRunner()
-    vendor_result = runner.invoke(
-        vendor,
-        ["--file", python_main_defaults_environment, "--solver", "mamba"],
-    )
-    assert vendor_result.exit_code == 0
-
-
-# test that the LockSpecification object returned from
-# get _lock_spec_for_environment_file contains the packages defined in
-# the python_conda_mirror_main_conda_forge_environment test fixture
-def test_get_lock_spec_for_environment_file(
-    python_conda_mirror_main_conda_forge_environment,
-):
-    from conda_vendor.conda_vendor import (
-        _parse_environment_file,
-        _get_conda_platform,
-    )
+    with runner.isolated_filesystem():
+        result = runner.invoke(vendor, ["--file", env, "--solver", solver])
+    assert result.exit_code == 0
 
-    lock_spec = _parse_environment_file(
-        python_conda_mirror_main_conda_forge_environment,
-        _get_conda_platform(),
-    )
-    assert any(
-        versioned_dep.name == "python"
-        for versioned_dep in lock_spec.dependencies
-    )
-    assert any(
-        versioned_dep.version == "3.9.5.*"
-        for versioned_dep in lock_spec.dependencies
-    )
 
+# tests the LockSpecification object returned from conda-lock actually has
+# the package we specified
+@pytest.mark.integration
+def test_get_lock_spec(make_env):
+    env = make_env(environments["minimal"])
+    lock_spec = _generate_lock_spec(env, _get_conda_platform())
+    for dep in lock_spec.dependencies:
+        if dep.name != "python":
+            continue
 
-# test that solve_environment's DryRunInstall object's
-# 'success' field is True
-def test_solve_environment(python_conda_mirror_main_conda_forge_environment):
+        if version.parse(dep.version) == version.parse("3.9.5"):
+            return
+
+    pytest.fail("python package not found in lock_spec")
 
-    env = Path(str(python_conda_mirror_main_conda_forge_environment))
-    solution = solve_environment(
-        env,
-        "conda",
-        "linux-64",
-    )
 
+# tests that the solve_environment call doesn't fail.  If this doesn't solve
+# a python Exception is raised
+@pytest.mark.integration
+def test_solvable_environment(make_env):
+    env = make_env(environments["conda_mirror"])
+    solution = solve_environment(env, "conda", "linux-64")
 
-def test_notsolvable_environment(python_conda_failing_env):
-    env = Path(str(python_conda_failing_env))
+
+# tests that if the sovle_environment call does fail, an Exception is raised
+@pytest.mark.integration
+def test_notsolvable_environment(make_env):
+    env = make_env(environments["notsolvable"])
     with pytest.raises(Exception):
```

Comment:
This doesn't check for a specific exception to be thrown?

Reply:
Yeah, the exception is thrown in `conda-lock` and I'm not sure what it will be, in general.

Reply:
This could be vulnerable to testing the wrong exception and still passing.

Reply:
there is no wrong exception.  I don't really understand your concern, can you spell it out?

Reply:
You are testing that an exception is thrown in the solve_environment function. How do you know the solve_environment function is failing in the manner you expect?

Reply:
if an exception is thrown, it is failing in the manner i expect.  That's what the test says and that's what I expect.  

I don't control the spot where the exception is thrown.  I don't care which exception is thrown I only hope that it inherits from Exception!  But if `conda-lock` changes its implementation this test will break.  But if `conda-lock` changes its implementation a lot of tests will break.

For now, I want the test to do what it does.